src/uint/div.rs

```diff
@@ -69,29 +74,8 @@ impl<const LIMBS: usize> Uint<LIMBS> {
     ///
     /// When used with a fixed `rhs`, this function is constant-time with respect
     /// to `self`.
-    #[allow(trivial_numeric_casts)]
     pub const fn div_rem_vartime(&self, rhs: &NonZero<Self>) -> (Self, Self) {
-        let mb = rhs.0.bits_vartime();
-        let mut bd = Self::BITS - mb;
-        let mut rem = *self;
-        let mut quo = Self::ZERO;
-        // If there is overflow, it means `mb == 0`, so `rhs == 0`.
-        let mut c = rhs.0.wrapping_shl_vartime(bd);
-
-        loop {
-            let (r, borrow) = rem.sbb(&c, Limb::ZERO);
-            let choice = ConstChoice::from_word_mask(borrow.0);
-            rem = Self::select(&r, &rem, choice);
-            quo = Self::select(&quo.bitor(&Self::ONE), &quo, choice);
-            if bd == 0 {
-                break;
-            }
-            bd -= 1;
-            c = c.shr1();
-            quo = quo.shl1();
-        }
-
-        (quo, rem)
+        div_rem_vartime(self, &rhs.0)
```

Comment:
Is there a particular reason you need a separate free function for `div_rem_vartime`, as opposed to placing the implementation here? They appear to have nearly the same type signature aside from supporting different lengths (which could also be changed here)

Reply:
I think just that the implementation could support a `div_rem_vartime_mixed` potentially. The signature for `Uint::div_rem_vartime` can't be made more flexible without creating type inference issues.

Reply:
> can't be made more flexible without creating type inference issues.

Really? Like what?

Most of the methods in `mul.rs` have a `const RHS_LIMBS: usize` without problems.

Reply:
Actually it's `rem_vartime` that creates a minor type inference issue in the montgomery params, but that's easily fixed: https://github.com/RustCrypto/crypto-bigint/blob/master/src/modular/monty_form.rs#L78-L80

Reply:
I moved the implementations inline and adjusted the signature of the vartime division methods, leaving the signature of `rem_vartime` alone for now.